core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1066,12 +1067,53 @@ private static RuntimeException notComparable(String op, Object b0,
         op, b1.getClass().toString()).ex();
   }
 
-  // &
-  /** Helper function for implementing <code>BIT_AND</code> */
+  // BITAND
+
+  /** SQL <code>BITAND</code> operator applied to long values */
   public static long bitAnd(long b0, long b1) {
     return b0 & b1;
   }
 
+  /** SQL <code>BITAND</code> operator applied to ByteString values */
+  public static ByteString bitAnd(ByteString b0, ByteString b1) {
+
+    if (b0.length() != b1.length()) {
+      throw RESOURCE.invalidInputForBitwiseBinaryValue().ex();
+    }
+    byte[] bytes0 = b0.getBytes();
+    byte[] bytes1 = b1.getBytes();
+    byte[] result = new byte[b0.length()];
```

Comment:
Technically speaking, `b0.getBytes()` clones the byte array, so there's no need to allocate `new byte[...]` for the result. I believe `bytes0` would work just fine for the resulting array.